aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -1425,10 +1425,6 @@ AT_ERROR("inverse: MAGMA library not found in "
   magma_int_t lda = std::max<magma_int_t>(1, n);
   magma_int_t lwork = n * magmaGetriOptimalBlocksize<scalar_t>(n);
 
-  // magmaLu and magmaGetri requires infos tensor to live on CPU
-  infos_lu = infos_lu.to(at::kCPU);
-  infos_getri = infos_getri.to(at::kCPU);
-
```

Comment:
@IvanYashchuk please review this in light of https://github.com/pytorch/pytorch/commit/a49406b33136a8c12503d00b1410e5596e19aede#r50939460

Reply:
Opened an issue to check other parts of code
https://github.com/pytorch/pytorch/issues/58769

Reply:
This change is not correct. `infos_lu` and `infos_getri` tensors are created on GPU by default. MAGMA calls wrapped in `magmaLu` and `magmaGetri` expect the info variables on host memory and will segfault with this change (checked with CUDA version < 10.1 and MAGMA 2.5.2). This segfault would be caught only when submitted from `ci-all/` branch, there is a CUDA 9 build IIRC. I've compiled the latest master branch of MAGMA on the ROCm platform and I see that there is no segfault. Apparently, the reason is that it's fine to dereference device pointers in host code on ROCm, but it's illegal on CUDA.
I'll submit an alternative fix that would work both for CUDA and ROCm.

Reply:
Moved this PR to draft mode; Will complete this PR once https://github.com/pytorch/pytorch/pull/59223 is done.

Reply:
Moving this PR out of draft mode.